packages/ssx-sdk/src/core.ts

```diff
@@ -193,8 +193,11 @@ export class SSXConnected {
       nonce: generateNonce(),
     };
 
-    const serverNonce = await this.ssxServerNonce(defaults);
-    if (serverNonce) defaults.nonce = serverNonce;
+    defaults.nonce = await this.ssxServerNonce(defaults);
```

Comment:
This would always overwrite the nonce [set in the line above](https://github.com/spruceid/ssx/pull/20/files#diff-92023938c9c08f9a708ddadca984236b869fbfe8b4622000e0fd4d4e3f859220R193) even if api is not set.